docs/developer-guide/ksqldb-reference/scalar-functions.md

```diff
@@ -734,6 +734,86 @@ multiple elements, like those containing wildcards, aren't supported.
 
     `CREATE STREAM LOGS (LOG STRUCT<CLOUD STRING, APP STRING, INSTANCE INT>, ...) WITH (VALUE_FORMAT='JSON', ...)`
 
+### `IS_JSON_STRING`
+
+Since: 0.25.0
+
+```sql
+is_json_string(json_string) -> Boolean
+```
+
+Given a string, returns `true` if it can be parsed as a valid JSON value, `false` otherwise.
+
+Examples:
+
+```sql
+is_json_string("[1, 2, 3]") => true
+is_json_string("{}") => true
+is_json_string("1") => true
+is_json_string("\"abc\"") => true
+is_json_string("null") => true
+is_json_string("") => false
+is_json_string("abc") => false
+is_json_string(NULL) => false
+```
+
+### `JSON_RECORDS`
+
+Since: 0.25.0
+
+```sql
+json_records(json_string) -> Array<Struct<json_key:String, json_value:String>>
+```
+
+Given a string, parses it as a JSON object and returns a ksqlDB array of structs containing 2 
```

Comment:
Ok, so this may be likely out-of-scope for this ticket, but I'm wondering if there are future functions which would produce JSON which should be inverse of this.  

E.g., I think there's interest in struct->json functions once other issues are resolved.  Have we thought through the consistency issues there?

Reply:
`CAST` is arguably the inverse of this. It does work for the primitive types. Containers are not yet supported, unfortunately:

https://github.com/confluentinc/ksql/blob/f7c1334a8b97f52db60f2e2ca84230c31c7e9da0/ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/helpers/CastEvaluator.java#L88-L96


Here is a test I tried to see how this will work out:

```sql
ksql> CREATE STREAM test (k STRING KEY, v INT)  WITH (kafka_topic='test_topic', value_format='JSON');
ksql> CREATE STREAM test (k STRING KEY, v INT)  WITH (kafka_topic='test_topic', value_format='JSON');

 Message
----------------
 Stream created
----------------
ksql> CREATE STREAM to_json AS SELECT k, to_json_STRING(v) AS JSON_v FROM test;

 Message
---------------------------------------
 Created query with ID CSAS_TO_JSON_17
---------------------------------------
ksql> insert into test (k, v) VALUES ('a', 1);
ksql> select * from to_json;
+-------------------------------------------------------------------------------+-------------------------------------------------------------------------------+
|K                                                                              |JSON_V                                                                         |
+-------------------------------------------------------------------------------+-------------------------------------------------------------------------------+
|a                                                                              |[1,2,3]                                                                        |
|a                                                                              |1                                                                              |
```

__NOTE__: the `[1,2,3]` is a leftover from a previous test that I didn't cut out to show how does an unsupported cast look.

```sql
ksql> select json_v, cast(json_v as int) as back_to_int from to_json;
+-------------------------------------------------------------------------------+-------------------------------------------------------------------------------+
|JSON_V                                                                         |BACK_TO_INT                                                                    |
+-------------------------------------------------------------------------------+-------------------------------------------------------------------------------+
|[1,2,3]                                                                        |null                                                                           |
|1                                                                              |1                                                                              |
```